xlators/mgmt/glusterd/src/glusterd-handshake.c

```diff
@@ -1867,44 +1867,40 @@ static rpc_clnt_prog_t gd_clnt_mgmt_hndsk_prog = {
 };
 
 static int
-glusterd_event_connected_inject(glusterd_peerctx_t *peerctx)
+glusterd_event_connected_inject(xlator_t *this, glusterd_peerctx_t *peerctx)
 {
-    GF_ASSERT(peerctx);
-
     glusterd_friend_sm_event_t *event = NULL;
+    glusterd_peerinfo_t *peerinfo = NULL;
     glusterd_probe_ctx_t *ctx = NULL;
     int ret = -1;
-    glusterd_peerinfo_t *peerinfo = NULL;
 
-    ret = glusterd_friend_sm_new_event(GD_FRIEND_EVENT_CONNECTED, &event);
-
-    if (ret) {
-        gf_msg("glusterd", GF_LOG_ERROR, 0, GD_MSG_EVENT_NEW_GET_FAIL,
-               "Unable to get new event");
-        goto out;
-    }
+    GF_ASSERT(peerctx);
 
     ctx = GF_CALLOC(1, sizeof(*ctx), gf_gld_mt_probe_ctx_t);
-
     if (!ctx) {
-        ret = -1;
-        gf_msg("glusterd", GF_LOG_ERROR, ENOMEM, GD_MSG_NO_MEMORY,
+        gf_msg(this->name, GF_LOG_ERROR, ENOMEM, GD_MSG_NO_MEMORY,
                "Memory not available");
-        goto out;
+        return ret;
     }
 
     RCU_READ_LOCK;
 
     peerinfo = glusterd_peerinfo_find_by_generation(peerctx->peerinfo_gen);
     if (!peerinfo) {
-        RCU_READ_UNLOCK;
```

Comment:
This should be considered as similar to a mutex, so it's much better to release it as soon as possible, specially if what we are going to do next is a log message which could take a significant amount of time.